test/unittests/test_xmlupload/test_stash_circular_references.py

```diff
@@ -0,0 +1,19 @@
+# pylint: disable=missing-class-docstring,missing-function-docstring,protected-access
+
+import pytest
+
+from dsp_tools.utils.xml_utils import parse_and_clean_xml_file
+from dsp_tools.utils.xmlupload.stash_circular_references import remove_circular_references
+from dsp_tools.utils.xmlupload.xmlupload import _extract_resources_from_xml
+
+test_root = parse_and_clean_xml_file("testdata/generate_test_data/test_circular_references_1.xml")
+resources = _extract_resources_from_xml(test_root, "simcir")
+
+
+def test_remove_circular_references() -> None:
+    ok_resources, _, _ = remove_circular_references(resources, False)
+    assert len(ok_resources) >= 26
```

Comment:
This is the length of the ok_resources. In the data resptr and standoff are mixed, if the code starts to remove different links than right now, one stash might become longer than it is right now, even if it removes less links in total. So that it is not necessary to add the two stashes together I chose to measure the length of the ok resources, which should get longer.

Reply:
ah... yea... clearly I was tired 🙈